src/envs/pybullet_blocks.py

```diff
@@ -0,0 +1,728 @@
+    # Fetch robot parameters.
+    _base_pose: Pose3D = (0.75, 0.7441, 0.0)
+    _base_orientation: Sequence[float] = [0., 0., 0., 1.]
+    _ee_orientation: Sequence[float] = [1., 0., -1., 0.]
+    _move_gain: float = 1.0
+    _max_vel_norm: float = 0.5
+    _grasp_offset_z: float = 0.01
+    _place_offset_z: float = 0.005
+    _grasp_tol: float = 0.05
+    _move_to_pose_tol: float = 0.0001
+    _finger_action_nudge_magnitude: float = 0.001
+
+    # Table parameters.
+    _table_pose: Pose3D = (1.35, 0.75, 0.0)
+    _table_orientation: Sequence[float] = [0., 0., 0., 1.]
+
+    # Block parameters.
+    _block_orientation: Sequence[float] = [0., 0., 0., 1.]
+    _block_mass = 0.5
+    _block_friction = 1.2
+    _block_colors: Sequence[Tuple[float, float, float, float]] = [
+        (0.95, 0.05, 0.1, 1.),
+        (0.05, 0.95, 0.1, 1.),
+        (0.1, 0.05, 0.95, 1.),
+        (0.4, 0.05, 0.6, 1.),
+        (0.6, 0.4, 0.05, 1.),
+        (0.05, 0.04, 0.6, 1.),
+        (0.95, 0.95, 0.1, 1.),
+        (0.95, 0.05, 0.95, 1.),
+        (0.05, 0.95, 0.95, 1.),
+    ]
+    _out_of_view_xy: Sequence[float] = [10.0, 10.0]
+
+    # Camera parameters.
+    _camera_distance: float = 0.8
+    _camera_yaw: float = 90.0
+    _camera_pitch: float = -24
+    _camera_target: Pose3D = (1.65, 0.75, 0.42)
+    _debug_text_position: Pose3D = (1.65, 0.25, 0.75)
+
+    def __init__(self) -> None:
+        super().__init__()
+
+        # Override options.
+        types = [self._robot_type, self._block_type]
+        params_space = Box(0, 1, (0, ))
+        self._Pick = utils.LinearChainParameterizedOption(
+            "Pick",
+            [
+                # Creates a ParameterizedOption for moving to the position that
+                # has z equal to self.pick_z, and x and y equal to that of the
+                # block object paramereter. In other words, move the end
+                # effector to high above the block in preparation for picking.
+                self._move_relative_to_block(
+                    "MoveToAboveBlock", (0., 0., self.pick_z),
+                    ("rel", "rel", "abs"), types, params_space),
+                # Open grippers.
+                self._change_grippers("OpenGrippers", self.open_fingers, types,
+                                      params_space),
+                # Move down to grasp.
+                self._move_relative_to_block(
+                    "MoveToGrasp", (0., 0., self._grasp_offset_z),
+                    ("rel", "rel", "rel"), types, params_space),
+                # Grasp.
+                self._change_grippers("Grasp", self.closed_fingers, types,
+                                      params_space),
+                # Move up.
+                self._move_relative_to_block(
+                    "MoveToAboveBlock", (0., 0., self.pick_z),
+                    ("rel", "rel", "abs"), types, params_space),
+            ])
+
+        types = [self._robot_type, self._block_type]
+        params_space = Box(0, 1, (0, ))
```

Comment:
Repeated from above? I can see why `params_space` might need to be a different object but `types` seems identical